### ISSUE TITLE
bug: ion-tab-bar hides when keyboard is opened with KeyboardResize.None

### ISSUE THREAD

**Issue by DereviankoAndrew** [ionic-team/ionic-framework#28953]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

ion-tab-bar gets class "tab-bar-hidden" when keyboard is opened, even through "resize: KeyboardResize.None" is set, which leads to a break of layout

https://github.com/ionic-team/ionic-framework/assets/59490330/2d26e0a8-24c2-45c1-9901-075bc464b17c

### Expected Behavior

ion-tab-bar shouldn't get class "tab-bar-hidden" when keyboard is opened and "resize: KeyboardResize.None" is set

### Steps to Reproduce

1. Add import of KeyboardResize from '@capacitor/keyboard' to capacitor.config.ts:
`import { KeyboardResize } from '@capacitor/keyboard';`
2. Add keyboard settings to plugins:
` plugins: {
    Keyboard: {
      resize: KeyboardResize.None,
    },
  }`
3. Try to focus on some ion-input, while being on page that contains "ion-tab-bar"
4. Layout will break because "ion-tab-bar" will get "display: none !important"

### Ionic Info

Ionic:

   Ionic CLI                     : 6.20.8 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : not installed
   @angular-devkit/build-angular : 17.1.1
   @angular-devkit/schematics    : 17.1.1
   @angular/cli                  : 17.1.1
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 5.0.5
   @capacitor/android : 5.0.5
   @capacitor/core    : 5.2.1
   @capacitor/ios     : 5.0.5

Utility:

   cordova-res                          : 0.15.4
   native-run (update available: 2.0.1) : 1.7.2

System:

   NodeJS : v18.16.1 (/usr/local/bin/node)
   npm    : 9.5.1
   OS     : macOS

### Additional Information

There was a pull request that unfortunately was not merged in October 2023: https://github.com/ionic-team/ionic-framework/pull/28310 , which made me mention this issue again.

Basically the problem is that in core/src/components/tab-bar/tab-bar.tsx constant "shouldHide" doesn't involve check of whether "KeyboardResize.None" is set, leading to the problem.

`render() {
    const { color, translucent, keyboardVisible } = this;
    const mode = getIonMode(this);
    const shouldHide = keyboardVisible && this.el.getAttribute('slot') !== 'top';

    return (
      <Host
        role="tablist"
        aria-hidden={shouldHide ? 'true' : null}
        class={createColorClasses(color, {
          [mode]: true,
          'tab-bar-translucent': translucent,
          'tab-bar-hidden': shouldHide,
        })}
      >
        <slot></slot>
      </Host>
    );
  }`

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as a duplicate of https://github.com/ionic-team/ionic-framework/issues/28226.
